python/ray/data/datasource/datasource.py

```diff
@@ -211,13 +212,30 @@ def __call__(self) -> Iterable[Block]:
 
         if context.block_splitting_enabled:
             for block in result:
-                yield block
+                yield from self._do_additional_splits(block)
         else:
             builder = DelegatingBlockBuilder()
             for block in result:
                 builder.add_block(block)
             yield builder.build()
 
+    def _set_additional_split_factor(self, k: int) -> None:
+        self._additional_output_splits = k
+
+    def _do_additional_splits(self, block: Block) -> Iterable[Block]:
+        if self._additional_output_splits > 1:
+            block = BlockAccessor.for_block(block)
+            offset = 0
+            split_sizes = np.array_split(
```

Comment:
Hmm, I don't know of a better method for this. I think this is probably going to be the fastest way since it's using numpy.

Reply:
Couldn't you compute the split size manually? Like `block.num_rows() // self._additional_output_splits` etc?

Reply:
It's a bit harder than that:

For an array of length l that should be split into n sections, it returns l % n sub-arrays of size l//n + 1 and the rest of size l//n.

I prefer to avoid the re-implementation risk here, as the overhead here will be dominated by the data copy and not this call. Splitting a million item array takes just a small fraction of a second.

Reply:
I was more concerned about the memory overhead, not the latency. This will be an extra 8 bytes per row, which may be significant if the rows are small. I don't think this is too hard or risky to implement.